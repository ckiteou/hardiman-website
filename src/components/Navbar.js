import React from 'react'
import navlinks from '../data/navbar.json'
import '../styles/nav.scss'

const Navbar = () => (
  <nav>
    <div className="nav_bar">
      <div className="logo">JMHGardens</div>
      <div className="nav_links_wrapper">
        {
          navlinks.map(i =>
            <a
              key={i.name}
              className="nav_link"
              href={i.link}
            >
              {i.name}
            </a>
          )
        }
      </div>
      <div className="contact_number">
        <i className="fas fa-phone-volume"/>
        &nbsp; 07932 771 158
      </div>
    </div>
  </nav>
)

export default Navbar;
