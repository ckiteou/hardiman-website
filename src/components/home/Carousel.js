import React, {Component} from 'react';
import { imgUrls }  from  '../../images/home/index.js'
import '../../styles/home.scss';


export const ImageSlide = ({url}) => {
  const styles = {
    backgroundImage: `url(${url})`,
  };
  return (
    <div className="image_slide" style={styles}></div>
  );
}

class Carousel extends Component {
  constructor (props) {
    super(props);
    this.state = {
      currentImageIndex: 0
    };
    this.setSlide = this.setSlide.bind(this);
    this.changeSlides = this.changeSlides.bind(this);
  }

  componentDidMount() {
    setTimeout(this.changeSlides, 5000);
  }

  changeSlides() {
    if (Number(this.state.currentImageIndex) < imgUrls.length - 1) {
      this.setState({
        currentImageIndex: Number(this.state.currentImageIndex) + 1
      });
    } else if (Number(this.state.currentImageIndex) + 1 === imgUrls.length) {
      this.setState({
        currentImageIndex: 0
      });
    }
    setTimeout(this.changeSlides, 5000);
  }

  setSlide(event) {
    let id = event.target.id;
    this.setState((state) => {
      return {currentImageIndex: state.currentImageIndex = id}
    });
  }

  render () {
    return (
      <div className="carousel">
        <ImageSlide url={imgUrls[this.state.currentImageIndex]}/>
        <h1 className="place_holder">PLACEHOLDER IMAGE</h1>
        <div className="dot_container">
          {
            imgUrls.map((v, i) => (
              <div
                key={i}
                className={Number(this.state.currentImageIndex) === i ? "dot active" : "dot"}
                onClick={this.setSlide}
                id={i}>
              </div>
            ))
          }
        </div>
      </div>
    );
  }
}

export default Carousel;
