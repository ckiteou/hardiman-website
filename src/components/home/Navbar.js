import React from 'react'
import { Link } from 'react-router'
import navlinks from '../../data/navbar.json'
import '../../styles/nav.scss'

export const ContactDetails = () => (
  <div className="contact_container">
    <p className="email">Email: jon_hardiman@xxxxx.com</p> {/*needs updating*/}
    <p className="number">Tel: 01235 666 xxx</p> {/*needs updating*/}
  </div>
)

const Navbar = () => (
  <nav>
    <div className="nav_bar">
      <h1 className="logo">LOGO</h1>{/*needs updating*/}
      {
        navlinks.map(i =>
          <Link key={i} className="nav_link" to="/">{i}</Link>)
      }
      <ContactDetails/>
    </div>
  </nav>
)

export default Navbar;
