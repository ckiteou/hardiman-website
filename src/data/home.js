export const personal_quote =
`"A garden is a reflction of our personalities, it's a place where we vision our own unique place to escape and relax.
That's why for me, it's important to build a relationship with my clients, to enable me to understand that vision and capture it in my work"`
