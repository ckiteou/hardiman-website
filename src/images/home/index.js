import tools from './tools.jpg'
import trough from './trough.jpg'

export const imgUrls = [
  tools,
  trough
]
