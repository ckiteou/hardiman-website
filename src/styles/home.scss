@import 'colours';

.home {
  height: 100%;

  .section_container {
    padding: 40px;
    .title-divider {
      display: flex;
      flex-direction: row;
      align-items: center;
      justify-content: center;
      text-align: center;
      margin-bottom: 20px;
      h2 {
        margin: 0;
        width: 25%;
        font-weight: normal;
        color: $primary;
      }
      hr {
        background-color: #e9e9e9;
        border: #e9e9e9;
        height: 1px;
        margin: 0;
        width: 37.5%;
      }    }
    .text {
      width: 70%;
    }
    img {
      width: 30%;
    }
  }

  .profile_image {
    position: relative;
    display: inline-block;
    box-sizing: border-box;
    width: 60%;
    left: 10%;
    height: auto;
    vertical-align: middle;
    border: solid 2px $primary;
    opacity: 0.9;
    box-shadow: 14px 14px 28px rgba(0,0,0,0.25), 7px 10px 10px rgba(0,0,0,0.22);
  }
}
