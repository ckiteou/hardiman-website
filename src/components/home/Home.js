import React, { Component } from 'react';
import Carousel from './Carousel'
import ScrollableAnchor, { configureAnchors } from 'react-scrollable-anchor'
import '../../styles/home.scss'
import '../../styles/hardiman.scss'

configureAnchors({offset: -70, scrollDuration: 800})

class Section extends Component {
  render() {
    return (
      <div className="section_container">
        <div className="title-divider">
          <hr/><h2>{this.props.title}</h2><hr/>
        </div>
        {this.props.children}
      </div>
    )
  }
}

// exract to seperate files
const About = () => (
  <div className="text" style={{height: '600px', padding: '20px'}}>About</div>
)

const Services = () => (
  <div className="text" style={{height: '600px', padding: '20px'}}>Services</div>
)

const Contact= () => (
  <div className="text" style={{height: '600px', padding: '20px'}}>Contact</div>
)

const Home = () => (
  <div className="home">
    <ScrollableAnchor id={'home'}>
      <Carousel/>
    </ScrollableAnchor>
    <ScrollableAnchor id={'about'}>
      <Section title="About">
        <About/>
      </Section>
    </ScrollableAnchor>
    <ScrollableAnchor id={'services'}>
      <Section title="Services">
        <Services/>
      </Section>
    </ScrollableAnchor>
    <ScrollableAnchor id={'contact'}>
      <Section title="Contact">
        <Contact/>
      </Section>
    </ScrollableAnchor>
  </div>
)

export default Home;
