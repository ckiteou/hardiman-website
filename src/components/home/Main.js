import React, { Component } from 'react';
import ScrollableAnchor, { configureAnchors } from 'react-scrollable-anchor'
import Services from '../Services'
import Contact from '../Contact'
import '../../styles/home.scss'
import '../../styles/hardiman.scss'

class Home extends Component {
  render() {
    return (
      <img src={'./images/watering.jpg'} style={{width: '100%', height: 'auto'}} alt="Watering Garden"/>
    )
  }
}

// exract to seperate files
const About = () => (
  <div className="text" style={{height: '600px', padding: '20px'}}>About</div>
)

configureAnchors({offset: -70, scrollDuration: 800})

class Section extends Component {
  render() {
    return (
      <div className="section_container">
        <div className="title-divider">
          <hr/><h2>{this.props.title}</h2><hr/>
        </div>
        {this.props.children}
      </div>
    )
  }
}

const Main = () => (
  <div className="home">
    <ScrollableAnchor id={'home'}>
      <Home/>
    </ScrollableAnchor>
    <ScrollableAnchor id={'about'}>
      <Section title="About">
        <About/>
      </Section>
    </ScrollableAnchor>
    <ScrollableAnchor id={'services'}>
      <Section title="Services">
        <Services/>
      </Section>
    </ScrollableAnchor>
    <ScrollableAnchor id={'contact'}>
      <Section title="Contact">
        <Contact/>
      </Section>
    </ScrollableAnchor>
  </div>
)

export default Main;
