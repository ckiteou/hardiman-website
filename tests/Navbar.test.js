import React from 'react';
import renderer from 'react-test-renderer'

import Navbar from '../src/components/home/Navbar';
import { ContactDetails } from '../src/components/home/Navbar';

describe('Navbar', () => {
  it('renders', () => {
    const tree = renderer.create(
      <Navbar />
    ).toJSON();
    expect(tree).toMatchSnapshot()
  });
});

describe('ContactDetails', () => {
  it('renders', () => {
    const tree = renderer.create(
      <ContactDetails />
    ).toJSON();
    expect(tree).toMatchSnapshot()
  });
});
