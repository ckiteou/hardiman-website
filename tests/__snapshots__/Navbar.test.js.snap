// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders 1`] = `
<nav>
  <div
    className="nav_bar"
  >
    <h1
      className="logo"
    >
      LOGO
    </h1>
    <a
      className="nav_link"
      href="#home"
    >
      Home
    </a>
    <a
      className="nav_link"
      href="#about"
    >
      About
    </a>
    <a
      className="nav_link"
      href="#services"
    >
      Services
    </a>
    <a
      className="nav_link"
      href="#contact"
    >
      Contact
    </a>
  </div>
</nav>
`;
